core/src/main/scala/kafka/server/Kip500Broker.scala

```diff
@@ -507,11 +507,12 @@ class Kip500Broker(val config: KafkaConfig,
     val (loadedBrokerMetadata, initialOfflineDirs) = getBrokerMetadataAndOfflineDirs
 
     /* check brokerId */
-    val loadedBrokerId = config.brokerId // TODO: grab loadedBrokerMetadata.brokerId when available in meta.properties
+    loadedBrokerMetadata.toProperties()
+    val loadedBrokerId = loadedBrokerMetadata.brokerId
     if (config.brokerId != loadedBrokerId)
       throw new InconsistentBrokerMetadataException(
         s"The configured Broker ID ${config.brokerId} doesn't match stored broker.id ${loadedBrokerId} in meta.properties. " +
-          s"The broker is trying to use the wrong log directory. Configured log.dirs may be wrong.")
+          s"The broker is trying to use the wrong log directory. Configured log.dirs or metadata.log.dir may be wrong.")
```

Comment:
if we're not going to be specific we should say something like "configured log directories may be wrong".  We shouldn't mention specific configurations unless we know it's that config, I think.

Reply:
Ack.